salt/grains/core.py

```diff
@@ -1695,9 +1695,9 @@ def os_data():
                                 grains['osfullname'] = "Leap"
                             elif os_release.get("VERSION") == "Tumbleweed":
                                 grains['osfullname'] = os_release["VERSION"]
-                        # Override VERSION_ID, if CPE_NAME around
-                        if cpe.get('version'):
-                            grains['lsb_distrib_release'] = cpe['version']
+                            # Override VERSION_ID, if CPE_NAME around
+                            if cpe.get('version') and cpe.get('vendor') == 'opensuse':  # Keep VERSION_ID for SLES
```

Comment:
As this is now moved inside the `elif`, we can remove the `cpe.get('version') and` part, as will be or 'suse' or 'opensuse' already.

Reply:
@aplanas no, because if there is no `version` due to broken CPE, the whole thing will still go wrong.